crates/metastore_client/src/session.rs

```diff
@@ -55,6 +56,31 @@ impl SessionCatalog {
         self.state.deployment.clone()
     }
 
+    pub fn resolve_native_table(
+        &self,
+        _database: &str,
+        schema: &str,
+        name: &str,
+    ) -> Option<&TableEntry> {
+        let schema_id = self.schema_names.get(schema)?;
+        let obj = self.schema_objects.get(schema_id)?;
+        let obj_id = obj.objects.get(name)?;
+
+        let ent = self
+            .state
+            .entries
+            .get(obj_id)
+            .expect("object name points to invalid id");
```

Comment:
should exit early if there is no entry instead of panicking.

```suggestion
        let ent = self
            .state
            .entries
            .get(obj_id)?;
```